src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,185 +1,96 @@
-import React, {Component } from "react";
-import {Product, NavigationBar, Cart} from "../../components";
+import React, { useEffect, useState } from "react";
+import { Product, NavigationBar, Cart, AddProduct } from "../../components";
 import { ProductApi } from "../../api";
-import { Modal, Button, InputNumber, } from "antd";
-import 'antd/dist/antd.css';
+import "antd/dist/antd.css";
 import styled from "styled-components";
 
-const loading = (
-  <div className="suspense-page">
-    {" "}
-    Loading...{" "}
-  </div>
-);
-
-export default class Catalogue extends Component {
-  state = {
-    menuOpened: false,
-    products: [],
-    selectedProduct: {},
-    onLoading: false,
-    showModal: false,
-    showModalCart: false,
-    cart: [],
-    qty: 1,
-  };
-
-  getDataProducts() {
-    return ProductApi.get().then(res => res.json())
-    .then(res => {
-      this.setState({
-        products : res,
-      })
-    });
+export default function NewCatalogue() {
+  const [products, setProducts] = useState([]);
+  const [selectedProduct, setSelectedProduct] = useState({});
+  const [showModalProduct, setShowModalProduct] = useState(false);
```

Comment:
The `selectedProduct` and `showModalProduct` are tied to each other. When you open the modal you always have a selected product, and when you close the modal you don't have a selected product. 

When different states are dependent on each other you can:
1. Use 2 (or more) `useState` hooks
2. Use 1 `useReducer` hook. See: https://reactjs.org/docs/hooks-reference.html#usereducer

Can you try to rewrite this to `useReducer`?